lib/msf/core/payload/linux/bind_tcp.rb

```diff
@@ -155,16 +141,9 @@ def asm_bind_tcp(opts={})
         mov al,0x3                    ; read syscall
         int 0x80                      ; invoke read
         xchg ebx,edi                  ; stash the accept socket in edi
-    ^
-    if close_socket
-      asm << %Q^
```

Comment:
FYI, I am always assuming that we're closing sockets now. We don't need to infer the socket state any more because the transport configuration tells Meterpreter what the original payload was.